### ISSUE TITLE
TXT records with periods in them are mistaken for multiple txt records

### ISSUE THREAD

**Issue by haakonnessjoen** [mdns-js/node-mdns-js#84]
As previously defined in closed issue #70:

TXT records are split if they have periods in the value part. This is allowed by spec, as all 8bit values should be allowed after =.

```
		this.advertisement = mdns.createAdvertisement(mdns.tcp('myproto'), 1234, {
			name: 'myproto-app',
			txt: {
				protocol_version: '1.0.0',
				network_uid: this.options.network_uid || '',
				application_name: this.options.applicationName
			}
		});
```

The resulting record is:
```
{ addresses: [ '10.0.0.x' ],
  query: [],
  port: 1234,
  fullname: 'myproto-app._myproto._tcp.local',
  txt:
   [ 'protocol_version=1',
     '0',
     '0',
     'network_uid=abcdef0123456',
     'application_name=my-testscript' ],
  type:
   [ { name: 'myproto',
       protocol: 'tcp',
       subtypes: [],
       description: undefined },
     { name: 'myproto',
       protocol: 'tcp',
       subtypes: [],
       description: undefined } ],
  host: 'myproto-app.local',
  interfaceIndex: 1,
  networkInterface: 'pseudo multicast' }
```

As you see, protocol version should be protocol_version=1.0.0, but instead it is incorrectly split up in multiple txt records.

**Comment by phatpaul** [comment]
I debugged the javascript, and the period is intact when the data is handed over to this.networking.send(packet);

Looking at the wireshark dump, the period is sent as a 0x04 (but should be ASCII 0x2E).
It seems 0x04 is used elsewhere in the protocol to separate parts like in:
myproto-app._myproto._tcp.local
The first period is sent as 0x04 and the second period is sent as 0x05.

I compared with wireshark the packet sent by a working mdns implementation vs. this one.  The working implementation correctly sends 0x2e for a period within a txt record.
I'll keep digging...

**Comment by phatpaul** [comment]
I traced it to another module node-dns-js, and it looks like you already fixed it.
https://github.com/mdns-js/node-dns-js/pull/11
How can I get that fix, do you have a fork I can use in npm?

**Comment by kmpm** [comment]
@phatpaul, could you test the code in PR #85.

**Comment by phatpaul** [comment]
Glad to see someone is still here.
I just tested PR #85 

I had previously fixed my issue with TXT records by doing:
`npm i --save-dev bitfocus/node-dns-js`

But to test your new PR I removed that one first.
```
npm r dns-js
npm i --save-dev mdns-js/node-mdns-js#pull/85/head
```
Your new code works as much as it did before, but still fails on the periods within TXT records.

BTW, https://github.com/mdns-js/node-dns-js/pull/11 did fix it.

**Comment by kmpm** [comment]
@phatpaul - please try again

**Comment by phatpaul** [comment]
Yes, that fixed the issue with TXT records.  But I'm still not able to get this to work with my application...   I'm going back to [node-mdns](https://github.com/agnat/node_mdns) and deal with the tricky install.
